src/main/java/uk/co/fivium/dmda/server/SMTPConfig.java

```diff
@@ -454,18 +470,25 @@ private int getUniqueChildNodeInt(Element pElement, String pChildTagName)
   }
 
   /**
-   * Returns the database configured for the given destination domain
+   * Returns the databases configured for the given destination domain
    *
    * @param pRecipientDomain Domain that maps to a database
-   * @return the database configured for the given destination domain
+   * @return a list of databases configured for the given destination domain
    */
-  public String getDatabaseForRecipient(String pRecipientDomain) {
-    for (DomainMatcher lMatcher : mRecipientDatabaseMapping){
-      if(lMatcher.match(pRecipientDomain)){
-        return lMatcher.getDatabase();
-      }
-    }
-    return null;
+  public List<String> getDatabasesForRecipient(String pRecipientDomain) {
+    List<DomainMatcher> lMatchedMatchers = mRecipientDatabaseMapping.stream()
+        .filter(matcher -> matcher.match(pRecipientDomain))
+        .collect(toList());
+
+    Integer lowestPriority = lMatchedMatchers.stream()
```

Comment:
It looks like this relies on `mRecipientDatabaseMapping` already being ordered, which is potentially error prone as someone may change the way `mRecipientDatabaseMapping` is initialised in the future and unknowingly break this logic.
It also seems like the order doesn't really matter, since you `filter` out based on priority below anyway?

Think it might be cleaner to just skip sorting `mRecipientDatabaseMapping` in the constructor and instead get the minimum priority value more semantically using `min` e.g.
```
int lowestPriority = lMatchedMatchers.stream()
        .map(DomainMatcher::getPriority)
        .min(Integer::compareTo)
        .orElse(Integer.MAX_VALUE); // seems like the default should be the 'lowest' priority to avoid accidentally filtering out other entries?
``` 

There might even be a way to get your `List<String>` in a single stream chain, but I can't think of a way after a minute of playing around. 
e.g. this isn't valid since `min()` returns a single value, rather than preserving duplicates
```
return lMatchedMatchers.stream()
        .min(Comparator.comparing(DomainMatcher::getPriority))
        .map(DomainMatcher::getDatabase)
        .collect(toList());
```